pkg/controller/controller.go

```diff
@@ -155,15 +159,45 @@ const (
 	errorWithParameters         string = "ErrorWithParameters"
 )
 
+// shouldReconcileBroker determines whether a broker should be reconciled; it
+// returns true if any of the following are true:
+//
+// - the broker's deletion timestamp is set
+// - the broker has no status conditions
+// - the broker has a ready condition whose status is true, and it has been
+//   longer than the given relist interval since the broker became ready
+func shouldReconcileBroker(broker *v1alpha1.Broker, now time.Time, relistInterval time.Duration) bool {
+	if broker.DeletionTimestamp != nil || len(broker.Status.Conditions) == 0 {
+		return true
+	}
+
+	for _, condition := range broker.Status.Conditions {
```

Comment:
Why loop through the conditions? Shouldn't we relist if it's not Ready as well?

The only case we shouldn't relist is ready =true and time is not enough. Everything else we need to go do something.